File: Python/SivitasAkademik.py
# Saya Ananda Myzza Marhelio NIM 2100702 mengerjakan soal Latihan 4 dalam mata kuliah Desain Pemrograman Berorientasi
# Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

from Human import Human


class SivitasAkademik(Human):  # Sivitas Akademik mewarisi Human.
    # Attribut private
    __asal_universitas = ""
    __email_edu = ""

    # Constructor
    def __init__(self, asal_universitas="", email_edu="", NIK="", name="", gender=""):
        super().__init__(NIK, name, gender)
        self.__asal_universitas = asal_universitas
        self.__email_edu = email_edu

    # Setter dan Getter
    # Setter & Getter untuk Asal Universitas.
    def getAsal_universitas(self):
        return self.__asal_universitas

    def setAsal_universitas(self, asal_universitas):
        self.__asal_universitas = asal_universitas

    # Setter & Getter untuk Email Edu.
    def getEmail_edu(self):
        return self.__email_edu

    def setEmail_edu(self, email_edu):
        self.__email_edu = email_edu

File: C++/SivitasAkademik.cpp
/* Saya Ananda Myzza Marhelio NIM 2100702 mengerjakan soal Latihan 4 dalam mata kuliah Desain Pemrograman Berorientasi
Objek untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. */

#include "Header.hh"

/* Constructor */
SivitasAkademik::SivitasAkademik() : Human()
{
    this->asal_universitas = "";
    this->email_edu = "";
}
SivitasAkademik::SivitasAkademik(string NIK, string name, string gender, string asal_universitas, string email_edu)
    : Human(NIK, name, gender)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
}

/* Setter & Getter untuk Asal Universitas */
string SivitasAkademik::getAsal_universitas()
{
    return this->asal_universitas;
}
void SivitasAkademik::setAsal_universitas(string asal_universitas)
{
    this->asal_universitas = asal_universitas;
}

/* Setter & Getter untuk Email Edu */
string SivitasAkademik::getEmail_edu()
{
    return this->email_edu;
}
void SivitasAkademik::setEmail_edu(string email_edu)
{
    this->email_edu = email_edu;
}

/* Destructor */
SivitasAkademik::~SivitasAkademik()
{
}
